### DISCUSSION TITLE
Recommended approach to ssh/scp into remote machine via a public jump host ip

### DISCUSSION THREAD

**Discussion by emailstorbala** [nushell/nushell#11836] [Q&A]
Currently I am trying to migrate an existing bash script to nushell. I am having difficulties in handling the below scenario.

I want to execute some command in a remote_host_ip which can be reached via jump_host_ip. Currently I am running the command as like below:

```
let SSH_PARAM = "-q -o BatchMode=yes -o ConnectTimeout=10 -o CheckHostIP=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
let proxy_cmd = $"ProxyCommand='ssh ($SSH_PARAM) -W %h:%p root@<jump_host>'"
let cmd = $"ssh -o ($proxy_cmd) root@<remote_host> 'hostnamectl 2>/dev/null'"
nu -c $cmd
```

The above code is framed from the existing bash command. Unfortunately I couldn't directly execute the resulting command but had to use 'nu -c $cmd' to execute the same.

Is there any better way to handle the same?

Similarly how to execute 'scp' of files into the remote_host_ip?

**Comment by fdncred** [comment]
You should be able to run the command in nushell if you set it up right. Making it one big string makes nushell think you have a command with spaces in it, which you don't. Typically, I do something like this.
```nushell
let cmd = "ssh"
let args = [-o ($proxy_cmd) root@<remote_host> 'hostnamectl 2>/dev/null']
$cmd $args
```
or you may need to tell it's an external command by prefixing $cmd with ^$cmd.

That's where I'd start.

**Reply by emailstorbala** [reply]
@fdncred I tried as suggested but couldn't make it work. Facing the below error.

![image](https://github.com/nushell/nushell/assets/12377900/d8b00a4e-d688-4e87-8109-95585f72b24e)

**Reply by fdncred** [reply]
It's hard for me to tell since I don't use ssh, but a "No such file or directory" error is indicative of it not finding something in your command line. What you need to do is determine what a proper command line looks like and them make the script create those values. I also just noticed that you have a ssh inside of a ssh with the proxy_cmd variable. It may not be a problem but I don't see that in your command line screenshot. I also didn't mention using run-external. It's probably not necessary.
